core/src/main/java/io/micronaut/core/annotation/AnnotationMetadata.java

```diff
@@ -1488,6 +1493,47 @@ default boolean hasDeclaredStereotype(@Nullable Class<? extends Annotation>... a
         return false;
     }
 
+    /**
+     * Is repeatable annotation?
+     * @param annotation The annotation
+     * @return true if repeatable
+     * @since 3.1
+     */
+    default boolean isRepeatableAnnotation(Class<? extends Annotation> annotation) {
+        return annotation.getAnnotation(Repeatable.class) != null;
+    }
+
+    /**
+     * Is repeatable annotation?
+     * @param annotation The annotation
+     * @return true if repeatable
+     * @since 3.1
+     */
+    default boolean isRepeatableAnnotation(String annotation) {
+        return false;
+    }
+
+    /**
+     * Find repeatable annotation container.
+     * @param annotation The annotation
+     * @return optional repeatable annotation container
+     * @since 3.1
+     */
+    default Optional<String> findRepeatableAnnotation(Class<? extends Annotation> annotation) {
+        return Optional.ofNullable(annotation.getAnnotation(Repeatable.class))
```

Comment:
Add nullability annotations to `annotation` parameter and if it can be `null` add null check or return `Optional.empty()` when `null`